storage/src/mmr/verification.rs

```diff
@@ -39,6 +39,75 @@ impl<H: CHasher> PartialEq for Proof<H> {
     }
 }
 
+impl<H: CHasher> EncodeSize for Proof<H> {
+    fn encode_size(&self) -> usize {
+        let digests_size = self.digests.len() * H::Digest::SIZE;
+        u64::SIZE + u16::SIZE + digests_size
+    }
+}
+
+impl<H: CHasher> Write for Proof<H> {
+    fn write(&self, buf: &mut impl BufMut) {
+        // Write the number of nodes in the MMR
+        self.size.write(buf);
```

Comment:
Consider wrapping `self.size` with `UInt(self.size)` in the Write/Read/EncodeSize functions to decrease the encoding size